src/login/login.component.tsx

```diff
@@ -10,10 +11,33 @@ export default function Login(props: LoginProps) {
   const [errorMessage, setErrorMessage] = React.useState("");
   const [isLoggingIn, setIsLoggingIn] = React.useState(false);
   const [showPassword, setShowPassword] = React.useState(false);
+  const [checkingIfLoggedIn, setCheckingIfLogged] = React.useState(true);
   const passwordInputRef = React.useRef<HTMLInputElement>(null);
   const usernameInputRef = React.useRef<HTMLInputElement>(null);
   const formRef = React.useRef<HTMLFormElement>(null);
 
+  React.useEffect(() => {
+    if (checkingIfLoggedIn) {
+      const subscription = getCurrentUser({
+        includeAuthStatus: true
+      }).subscribe(
+        authResult => {
+          if (authResult.authenticated) {
```

Comment:
If you update getCurrentUser to only return the user object, it will not have a authenticated property. Furthermore, I think that if we need access to the session properties, we should have a separate api for accessing the session info rather than piggybacking on getCurrentUser

Reply:
Nevermind. I see the change you made to getCurrentUser.
LGTM

Reply:
The advantage of using `getCurrentUser` is sharing the network request -- it's the exact same url for both checking auth status and getting the user.